client/analytics/report/orders/table.js

```diff
@@ -0,0 +1,267 @@
+import { format as formatDate } from '@wordpress/date';
+import { map, orderBy } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { Card, TableCard, TablePlaceholder } from '@woocommerce/components';
+import { downloadCSVFile, generateCSVDataFromTable, generateCSVFileName } from 'lib/csv';
+import { formatCurrency, getCurrencyFormatDecimal } from 'lib/currency';
+import { getIntervalForQuery, getDateFormatsForInterval } from 'lib/date';
+import { getAdminLink, onQueryChange } from 'lib/nav-utils';
+
+export default class OrdersReportTable extends Component {
+	constructor( props ) {
+		super( props );
+	}
+
+	onDownload( headers, rows, query ) {
+		// @TODO The current implementation only downloads the contents displayed in the table.
+		// Another solution is required when the data set is larger (see #311).
+		return () => {
+			downloadCSVFile(
+				generateCSVFileName( 'orders', query ),
+				generateCSVDataFromTable( headers, rows )
+			);
+		};
+	}
+
+	getHeadersContent() {
+		return [
+			{
+				label: __( 'Date', 'wc-admin' ),
+				key: 'date_created',
+				required: true,
+				defaultSort: true,
+				isIdentifier: true,
+				isSortable: true,
+			},
+			{
+				label: __( 'Order #', 'wc-admin' ),
+				key: 'id',
+				required: true,
+				isIdentifier: true,
+				isSortable: true,
+			},
+			{
+				label: __( 'Status', 'wc-admin' ),
+				key: 'status',
+				required: false,
+				isSortable: true,
+			},
+			{
+				label: __( 'Customer', 'wc-admin' ),
+				key: 'customer_id',
+				required: false,
+				isSortable: true,
+			},
+			{
+				label: __( 'Product(s)', 'wc-admin' ),
+				key: 'products',
+				required: false,
+				isSortable: false,
+			},
+			{
+				label: __( 'Items Sold', 'wc-admin' ),
+				key: 'items_sold',
+				required: false,
+				isSortable: true,
+				isNumeric: true,
+			},
+			{
+				label: __( 'Coupon(s)', 'wc-admin' ),
+				key: 'coupons',
+				required: false,
+				isSortable: false,
+			},
+			{
+				label: __( 'N. Revenue', 'wc-admin' ),
+				key: 'net_revenue',
+				required: false,
+				isSortable: true,
+				isNumeric: true,
+			},
+		];
+	}
+
+	formatTableData( data ) {
+		return map( data, row => {
+			const {
+				date_created,
+				id,
+				status,
+				customer_id,
+				line_items,
+				coupon_lines,
+				currency,
+				total,
+				total_tax,
+				shipping_total,
+				discount_total,
+			} = row;
+
+			return {
+				date_created,
+				id,
+				status,
+				customer_id,
+				line_items,
+				items_sold: line_items.reduce( ( acc, item ) => item.quantity + acc, 0 ),
+				coupon_lines,
+				currency,
+				net_revenue: getCurrencyFormatDecimal(
+					total - total_tax - shipping_total - discount_total
+				),
+			};
+		} );
+	}
+
+	getRowsContent( tableData ) {
+		const { query } = this.props;
+		const currentInterval = getIntervalForQuery( query );
+		const { tableFormat } = getDateFormatsForInterval( currentInterval );
+		const statusNames = {
```

Comment:
Can we leverage `wcSettings.orderStatuses` here instead of this hard-coded list? That array also provides support for custom order statuses that might be added by extensions.